### PR TITLE
feat: migrate to sage and upgrade to go 1.19

### PR DESCRIPTION
**Pull request by hashemmm96**
- Remove tools dir
- Fix linting errors
- Use go:generate for code generation and move mocks
- Update workflows to use go-semantic-release
- Change dependabot update frequency from daily to weekly

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -1,29 +1,267 @@
 run:
```
Comment (odsod) [inline .golangci.yml:1]: Why not use the bundled config?

Comment (hashemmm96) [reply .golangci.yml:1]: I assumed the config in the Sage repo would be more up-to-date. Is that not the case?

Comment (odsod) [reply .golangci.yml:1]: The config in the Sage repo is `go:embeded` and automatically used by the `sggolangcilint.Run` helper, so you don't need to bring it unless you want to customize it (not recommended)

Comment (hashemmm96) [reply .golangci.yml:1]: Or maybe you meant why not remove the file completely? That is definitively better

File: pkg/candebug/http_test.go [hunk 2]
```diff
@@ -23,9 +25,19 @@ func TestServeMessagesHTTP_Single(t *testing.T) {
 			},
 		})
 	}))
-	res, err := http.Get(ts.URL)
+	c := http.Client{
```
Comment (alethenorio) [inline pkg/candebug/http_test.go:28]: Is this for wanting to set 30 second timeout? Is that needed on the test? Wondering whether we can simplify by using the default client instead.

File: pkg/socketcan/emulator.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ type EmulatorOption func(*emulatorCfg)
 
 // WithMulticastAddress sets the address for the multicast group that the Emulator should listen on.
 // A multicast address starts with 239.x.x.x, and using an address that does not conform to this
-// will lead to undefined behaviour.
```
Comment (alethenorio) [inline pkg/socketcan/emulator.go:35]: I often tend to spell it the English way as I find it more common. Is this not considered the most common way?

File: tools.go [hunk 4]
```diff
@@ -1,8 +0,0 @@
-// +build tools
-
-package tools
-
-import (
-	_ "github.com/golang/mock/mockgen"
-	_ "golang.org/x/tools/cmd/stringer"
```
Comment (alethenorio) [inline tools.go:7]: With this deleted, what prevents these tools from being removed if you run go mod tidy? Won't this cause the these packages to be removed from the go.mod which in turn causes sage targets to fail?

**Review by alethenorio — APPROVED** [review]
Some minor comments but looks good in general. Nice work

File: pkg/candebug/http_test.go (hunk 2 above)
Comment (hashemmm96) [reply pkg/candebug/http_test.go:28]: The linter complains when you use http.Get without a context. An option would be to disable linter for these cases in tests.

File: pkg/socketcan/emulator.go (hunk 3 above)
Comment (hashemmm96) [reply pkg/socketcan/emulator.go:35]: Linter complaints as well.

File: tools.go (hunk 4 above)
Comment (hashemmm96) [reply tools.go:7]: These tools are installed through sage in the .sage/tools repo instead, making them only a build-time dependency and not a runtime dependency. We only use them when building and running test so we don't want them as runtime dependencies.

File: pkg/candebug/http_test.go (hunk 2 above)
Comment (alethenorio) [reply pkg/candebug/http_test.go:28]: Got it. My thinking was to replace this with `http.DefaultClient` for simplicity

File: README.md [hunk 5]
```diff
@@ -1,39 +1,29 @@
...
-[![GoReportCard][report-badge]][report]
-[![Codecov][codecov-badge]][codecov]
-
-[pkg-badge]: https://pkg.go.dev/badge/go.einride.tech/can
-[pkg]: https://pkg.go.dev/go.einride.tech/can
-[report-badge]: https://goreportcard.com/badge/go.einride.tech/can
-[report]: https://goreportcard.com/report/go.einride.tech/can
-[codecov-badge]: https://codecov.io/gh/einride/can-go/branch/master/graph/badge.svg
-[codecov]: https://codecov.io/gh/einride/can-go
+[![PkgGoDev](https://pkg.go.dev/badge/go.einride.tech/can)](https://pkg.go.dev/go.einride.tech/can) [![GoReportCard](https://goreportcard.com/badge/go.einride.tech/can)](https://goreportcard.com/report/go.einride.tech/can) [![Codecov](https://codecov.io/gh/einride/can-go/branch/master/graph/badge.svg)](https://codecov.io/gh/einride/can-go)
```
[the remark this answers is not in the document]
Comment (hashemmm96) [reply README.md:4]: Yes, totally agree on the readability. But this seems to only affect the readability for badges, so I think it's OK to leave it like that.

File: tools.go (hunk 4 above)
Comment (alethenorio) [reply tools.go:7]: Yes but sage is doing "InstallWithModFile` which reads the version from the dependency in the go.mod file and if these tools are not defined in the go.mod how is that working?

Unless I am missing something I think we need to either change sage to use the latest version (and not depend on the go.mod file) or keep this tools.go

Comment (hashemmm96) [reply tools.go:7]: Now I see what you mean. It has only worked because we already depend on these packages from other sources. I will re-add this file.

**Pull request merged**